### ISSUE TITLE
Submit job request processing entire bucket.

### ISSUE THREAD

**Issue by LeoReubelt** [aws/aws-sdk-js-v3#6157]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

I have an application that uses Health Imaging.
And I have a lambda that starts import jobs with the following function.

```
         const command = new StartDICOMImportJobCommand({
            clientToken: uuid.v4(),
            inputOwnerAccountId: "",
            datastoreId: dataStoreId,
            dataAccessRoleArn:
                'arn:aws:iam::' +
                process.env.ACCOUNT +
                ':role/ADD_DICOM_TO_DATASTORE_ROLE-' +
                process.env.ENV,
            inputS3Uri: inputPath,
            outputS3Uri: outputPath,
            jobName: awsRequestId,
         });
        
        const response = await client.send(command);
```

I enter a specific folder for inputS3Uri.
And it has been working great.
But I started getting the following error.

```
DICOM import job has failed.
FILE_COUNTS_LIMIT_EXCEED: Input bucket contains more than 5000 files.
```
And it seems like it is processing the entire bucket. Not just the folder that I pass in.
If I go to the Data Store import job in the console and click on the link for the 'Import source location in S3', i get the following error...

`The bucket named "xxxxxx" can't be found in the US East (Ohio) us-east-2 Region. The bucket named "xxxxx" is in the US East (N. Virginia) us-east-1 Region.`

And then it redirects to the correct bucket, but not to the folder specified and not to the root, but to a folder in between.
I have confirmed that the path is correct and that the files at the path exist. 
Also everything in our application is us-east-1. There are no references to us-east-2. And I am explicitly specifying us-east-1 when the clients are created. 

```
const s3 = new AWS.S3({
    region: process.env.REGION,
});
const ahi = new AWS.MedicalImaging({
    region: process.env.REGION,
});
```

So, why is health imaging processing the entire bucket, and where is it getting us-east-2 from?
I also tried this with the v2 sdk and got the same results.

### SDK version number

@aws-sdk/client-medical-imaging@3.414.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v16.14.0

### Reproduction Steps

See above

### Observed Behavior

Specifying a folder in an s3 bucket that contains files to submit as a health imaging import job does not work. More files than are actually in that folder are submitted.

### Expected Behavior

Only the files in the specified folder would be submitted.

**Comment by aBurmeseDev** [comment]
Hi @LeoReubelt - thanks for reaching out. 

I attempted to reproduce it with the similar code below and was unable to. I passed the region into client as you can see. Here's [order of precedence](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/setting-region.html#setting-region-order-of-precedence) for setting the region in v3. I would suggest double checking lambda function, your env variables where region is set and lastly try setting region in client constructor.

```ts
import { MedicalImagingClient, StartDICOMImportJobCommand } from "@aws-sdk/client-medical-imaging";
const client = new MedicalImagingClient({region: "us-west-2"});
const input = { 
  jobName: "STRING_VALUE",
  dataAccessRoleArn: "STRING_VALUE",
  clientToken: "STRING_VALUE",
  datastoreId: "STRING_VALUE",
  inputS3Uri: "STRING_VALUE", 
  outputS3Uri: "STRING_VALUE", 
  inputOwnerAccountId: "STRING_VALUE",
};
const command = new StartDICOMImportJobCommand(input);
const response = await client.send(command);
```

Let me know if issue persists and I'd be happy to further look into it.

**Comment by LeoReubelt** [comment]
@aBurmeseDev - To be clear, the issue the issue that i am trying to solve, is the `DICOM import job has failed. FILE_COUNTS_LIMIT_EXCEED: Input bucket contains more than 5000 files` Error.

I have customers that can't submit their data and I have a support ticked entered with AWS support that is now 25 days with no response.

Previously when I checked the image sets, the updated date for almost all of the image sets was the date of my most recent submission. And I was certainly not submitting 5000 files so it definitely was not respecting the folder path that I was passing in. 

I deleted the previously processed files, so the site was working again. And upgraded the sdk version. After the upgrade, the updated date was correct. Image set updated dates reflected the date they were submitted. Not the date of the last submission. So I thought it was fixed.

But last night, users could not submit again. Getting the same error...
DICOM import job has failed. FILE_COUNTS_LIMIT_EXCEED: Input bucket contains more than 5000 files.

So my question is, can I only have 5000 files total in the bucket (not folder) that I am using for dicom processing input? Or is something else going on here?
